activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -64,7 +64,23 @@ def self.build_mem_cache(*addresses) # :nodoc:
         addresses = addresses.flatten
         options = addresses.extract_options!
         addresses = ["localhost:11211"] if addresses.empty?
-        Dalli::Client.new(addresses, options)
+
+        pool_options = {}
+        pool_options[:size] = options[:pool_size] if options[:pool_size]
+        pool_options[:timeout] = options[:pool_timeout] if options[:pool_timeout]
```

Comment:
What are the defaults for these? If neither is set, you won't get any pooling, but if one is set, the other will presumably adopt its ConnectionPool default... is that the most predictable behaviour? (Specifically, I'm assuming there's a default timeout, and wondering whether it's more intuitive for us to respect that, or to apply an infinite timeout [more like the unpooled case] when someone only specifies a size.)

Reply:
Well, both have the defaults of 5 and if you specify one and not specify the other, the default will be assumed, see https://github.com/mperham/connection_pool/blob/6dea06663c80cf50cd6b5a410376943d328b4f1e/lib/connection_pool.rb#L47.

I think that an infinity wait is worse than defaulting to 5 because it would lock a web server thread that is not always easy to detect, a screaming exception is always easier to detect.

If you don't specify any of the two, the connection pool won't be used, as is today.

Wdyt?